File: stack.py
class Node:
    def __init__(self, item, next):
        self.item = item  # value
        self.next = next  # next node 타입이 다름


class Stack:
    def __init__(self):
        self.last = None

    def push(self, item):
        self.last = Node(item, self.last)

    def pop(self):
        last = self.last.item
        self.last = self.last.next
        return last

    def print(self):
        element = self.last
        while element is not None:
            print(f"{element.item}")
            element = element.next


stack = Stack()
stack.push(5)
stack.push(4)
stack.push(3)
stack.print()

File: stack.rs
struct Node {
    item: i32,
    next: Option<Box<Node>>,
}

impl Node {
    fn new(item: i32, next: Option<Box<Node>>) -> Node {
        Node { item, next }
    }
}

struct Stack {
    last: Node,
}

impl Stack {
    fn new() {}
    fn push() {}
    // fn pop() -> Node {}
}

fn main() {
    let node1 = Node::new(5, None);
    let node2 = Node::new(10, Some(Box::new(node1)));
    let node3 = Node::new(7, Some(Box::new(node2)));

    // 리스트 출력
    let mut current = Some(Box::new(node3));
    while let Some(node) = current {
        println!("{}", node.item);
        current = node.next;
    }
}
